pkg/controller/workload/kubernetes/resource/resource.go

```diff
@@ -400,21 +399,21 @@ type clusterConnecter struct {
 
 func (c *clusterConnecter) config(ctx context.Context, ar *v1alpha1.KubernetesApplicationResource) (*rest.Config, error) {
 	n := types.NamespacedName{Namespace: ar.GetNamespace(), Name: ar.GetName()}
-	if ar.Status.Cluster == nil {
+	if ar.Status.Target == nil {
 		return nil, errors.Errorf("%s %s is not scheduled", v1alpha1.KubernetesApplicationResourceKind, n)
 	}
 
-	n = types.NamespacedName{Namespace: ar.GetNamespace(), Name: ar.Status.Cluster.Name}
-	k := &computev1alpha1.KubernetesCluster{}
+	n = types.NamespacedName{Namespace: ar.GetNamespace(), Name: ar.Status.Target.Name}
+	k := &v1alpha1.KubernetesTarget{}
 	if err := c.kube.Get(ctx, n, k); err != nil {
-		return nil, errors.Wrapf(err, "cannot get %s %s", computev1alpha1.KubernetesClusterKind, n)
+		return nil, errors.Wrapf(err, "cannot get %s %s", v1alpha1.KubernetesTargetKind, n)
 	}
 
-	if k.Spec.WriteConnectionSecretToReference == nil {
-		return nil, errors.Errorf("%s %s has no connection secret", computev1alpha1.KubernetesClusterKind, n)
+	if k.Spec.ConnectionSecretRef == nil {
+		return nil, errors.Errorf("%s %s has no connection secret", v1alpha1.KubernetesTargetKind, n)
 	}
 
-	n = types.NamespacedName{Namespace: k.GetNamespace(), Name: k.Spec.WriteConnectionSecretToReference.Name}
+	n = types.NamespacedName{Namespace: k.GetNamespace(), Name: k.Spec.ConnectionSecretRef.Name}
```

Comment:
I'd go with using `GetWriteConnectionSecretToReference() *v1alpha1.LocalSecretReference` for name here.